core/shared/src/main/scala/fs2/async/mutable/Queue.scala

```diff
@@ -40,12 +40,22 @@ trait Queue[F[_], A] { self =>
   /** Dequeue one `A` from this queue. Completes once one is ready. */
   def dequeue1: F[A]
 
+  /** Dequeue all available `A`s from this queue. Completes once any values are ready. */
+  def dequeueAvailable1: F[Vector[A]]
+
   /** Like `dequeue1` but provides a way to cancel the dequeue. */
   def cancellableDequeue1: F[(F[A], F[Unit])]
 
+  /** Like `dequeueAvailable1` but provides a way to cancel the dequeue. */
+  def cancellableDequeueAvailable1: F[(F[Vector[A]], F[Unit])]
+
   /** Repeatedly call `dequeue1` forever. */
   def dequeue: Stream[F, A] = Stream.bracket(cancellableDequeue1)(d => Stream.eval(d._1), d => d._2).repeat
 
+  /** Repeatedly call `dequeueAvailable1` forever. */
+  def dequeueAvailable: Stream[F, A] =
```

Comment:
Whats wrong with `constant(Int.MaxValue).through(dequeueAvailable)` to get all available elements? Also if that seems ugly, we can introduce `dequeueAllAvailable` as an alias for this. Also I have no intention to introduce any dedicated queue bound to that pipe. This is simply to add the parameter to dequeue operation that will say how much element you are interested in instead on relying only on defaults (1) or (All). The possibility of abstract this for both cases is just free gain to me not the original intention. 

The idea of flow-control is something you may or may not use in this case. Imagine situation where you would like to feed the results to some network I/O laters. Maybe some of the I/O pipes may be slower then others, while still we want to maximize the performance to take always all the elements we can process in given time frame in single chunk. In that case the max elements may really depend on two factors, time and speed of the outbound I/O. Wouldn't it be nice in that case to dynamically bound the elements available to be dequeued?